File: c_src/26.c
#include "../include/my.h"

int removeDuplicates(int* nums, int numsSize){
    if (numsSize < 2) {
        return numsSize;
    }
    int i, j;
    for (i = 0, j = 1; j < numsSize; j++) {
        if (nums[i] != nums[j]) {
            nums[++i] = nums[j];
        }
    }

    return i;
}

File: go_src/26.go
func removeDuplicates(nums []int) int {
    i, n := 0, len(nums)
    if n < 2 {
        return n
    }
    for j := 1; j < n; j++ {
        if nums[i] != nums[j] {
            i++
            nums[i] = nums[j]
        }
    }

    return i + 1
}
